server/lib/src/constants/acp.rs

```diff
@@ -2,336 +2,475 @@
+                        ]}
+                    }
+                ]}"#
             )
-                .expect("Invalid JSON filter")
+            .expect("Invalid JSON filter")
         ),
-        ("acp_modify_removedattr", Value::new_iutf8("name")),
-        ("acp_modify_removedattr", Value::new_iutf8("displayname")),
-        ("acp_modify_removedattr", Value::new_iutf8("legalname")),
-        ("acp_modify_removedattr", Value::new_iutf8("mail")),
-
-        ("acp_modify_presentattr", Value::new_iutf8("name")),
-        ("acp_modify_presentattr", Value::new_iutf8("displayname")),
-        ("acp_modify_presentattr", Value::new_iutf8("legalname")),
-        ("acp_modify_presentattr", Value::new_iutf8("mail"))
+        (ATTR_ACP_MODIFY_REMOVEDATTR, Attribute::Name.to_value()),
+        (
+            ATTR_ACP_MODIFY_REMOVEDATTR,
+            Attribute::DisplayName.to_value()
+        ),
+        (ATTR_ACP_MODIFY_REMOVEDATTR, Attribute::LegalName.to_value()),
+        (ATTR_ACP_MODIFY_REMOVEDATTR, Attribute::Mail.to_value()),
+        (ATTR_ACP_MODIFY_PRESENTATTR, Attribute::Name.to_value()),
+        (
+            ATTR_ACP_MODIFY_PRESENTATTR,
+            Attribute::DisplayName.to_value()
+        ),
+        (ATTR_ACP_MODIFY_PRESENTATTR, Attribute::LegalName.to_value()),
+        (ATTR_ACP_MODIFY_PRESENTATTR, Attribute::Mail.to_value())
     );
 }
 
 lazy_static! {
     pub static ref E_IDM_ACP_PEOPLE_MANAGE_PRIV_V1: EntryInitNew = entry_init!(
-        ("class", CLASS_OBJECT.clone()),
-        ("class", CLASS_ACCESS_CONTROL_PROFILE.clone()),
-        ("class", CLASS_ACCESS_CONTROL_DELETE.clone()),
-        ("class", CLASS_ACCESS_CONTROL_CREATE.clone()),
-        ("name", Value::new_iname("idm_acp_people_manage")),
-        ("uuid", Value::Uuid(UUID_IDM_ACP_PEOPLE_MANAGE_PRIV_V1)),
-        (
-            "description",
+        (ATTR_CLASS, EntryClass::Object.to_value()),
+        (ATTR_CLASS, EntryClass::AccessControlProfile.to_value()),
+        (ATTR_CLASS, EntryClass::AccessControlDelete.to_value()),
+        (ATTR_CLASS, EntryClass::AccessControlCreate.to_value()),
+        (ATTR_NAME, Value::new_iname("idm_acp_people_manage")),
+        (ATTR_UUID, Value::Uuid(UUID_IDM_ACP_PEOPLE_MANAGE_PRIV_V1)),
+        (
+            Attribute::Description.as_ref(),
             Value::new_utf8s("Builtin IDM Control for creating person (user) accounts")
         ),
         (
-            "acp_receiver_group",
+            ATTR_ACP_RECEIVER_GROUP,
             Value::Refer(UUID_IDM_PEOPLE_MANAGE_PRIV)
         ),
         (
-            "acp_targetscope",
+            ATTR_ACP_TARGET_SCOPE,
             Value::new_json_filter_s(
-                "{\"and\": [{\"eq\": [\"class\",\"account\"]}, {\"eq\": [\"class\",\"person\"]}, {\"andnot\": {\"or\": [{\"eq\": [\"memberof\",\"00000000-0000-0000-0000-000000001000\"]}, {\"eq\": [\"class\", \"tombstone\"]}, {\"eq\": [\"class\", \"recycled\"]}]}}]}"
+                r#"{"and": [
```

Comment:
I'm a fan of this multi-line expansion.

Reply:
You'll be a bigger fan when the JSON is gone 😄